### PR TITLE
EWPP-2724: Bubble up facet cache.

### PR DESCRIPTION
**Pull request by nagyad**

### CODE REVIEW

File: src/EventSubscriber/QuerySubscriber.php [hunk 1]
```diff
@@ -130,6 +130,10 @@ public function queryAlter(QueryPreExecuteEvent $event) {
         'facet' => $facet,
       ]);
       $query_type_plugin->execute();
+      // Add query tags to the query.
```
Comment (upchuk) [inline src/EventSubscriber/QuerySubscriber.php:133]: Query tags? Also, if you are gonna put a comment, say why you are adding these tags.

I see in multiple places you are adding the cache tags of the Facet entity to things. What relevant facet cache tags do we need? Do we need to invalidate any caches whenever a facet gets changed (which happens only as a deployment action)? Or are there tags I don't know about?

**Review by upchuk — CHANGES_REQUESTED** [review]
I see no tests for any of these changes.

File: tests/src/Kernel/ListBuilderTest.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    $entityRepository = $this->container->get('entity.repository');
+    $formBuilder = $this->container->get('form_builder');
+    $facetsUrlGenerator = $this->container->get('facets.utility.url_generator');
+    $processorManager = $this->container->get('plugin.manager.facets.processor');
+    $requestStack = $this->container->get('request_stack');
+    $urlProcessorManager = $this->container->get('plugin.manager.facets.url_processor');
+    $multiselectFilterManager = $this->container->get('plugin.manager.multiselect_filter_field');
+    $listSourceFactory = $this->container->get('oe_list_pages.list_source.factory');
+
+    $listBuilder = new ListBuilder($listExecutionManager, $this->entityTypeManager, $pager, $entityRepository, $formBuilder, $facetsUrlGenerator, $processorManager, $requestStack, $urlProcessorManager, $multiselectFilterManager, $listSourceFactory);
```
Comment (hernani) [inline tests/src/Kernel/ListBuilderTest.php:50]: $listBuilder = $this->container->get('oe_list_pages.builder'); 

All the remaining services don't need to be instantiated.

File: tests/src/Kernel/DateStatusTest.php [hunk 3]
```diff
@@ -71,18 +71,22 @@ public function testWithConfiguredDefaults(): void {
       DateStatus::UPCOMING => t('Upcoming'),
       DateStatus::PAST => t('Past'),
     ];
+    // Assert the query has the corresponding cache tag from the facet.
+    $this->assertContains($this->facet->getCacheTags()[0], $query->getCacheTags());
```
Comment (hernani) [inline tests/src/Kernel/DateStatusTest.php:75]: Compare this against a static list of cache tags and compare both are similar.

**Review by hernani — CHANGES_REQUESTED** [review]
Changes requested

File: docker-compose.yml [hunk 4]
```diff
@@ -15,7 +15,7 @@ services:
       XDEBUG_MODE: "develop, debug"
       PHP_IDE_CONFIG: "serverName=Docker"
       # Enable step debugging for all PHP request. See ./README.md#step-debugging for more information.
-      # XDEBUG_SESSION: 1
+      XDEBUG_SESSION: 1
```
Comment (hernani) [inline docker-compose.yml:18]: ?

**Review by hernani — APPROVED** [review]

**Review by upchuk — APPROVED** [review]

**Pull request merged**